food-truck-frontend/src/components/login/LoginComponent.tsx

```diff
@@ -0,0 +1,72 @@
+import React from 'react'
+import Form from "../Form";
+import {AxiosResponse} from 'axios';
+import {Grid, TextField, Typography} from '@material-ui/core'
+import {UserData} from "../../redux/user/UserReducer";
+import NextLink from 'next/link'
+import LinkButton from "../layout/LinkButton";
+import Router from "next/router";
+
+type State = {
+  resultText: string;
+  data: UserData;
+  loadUserFromBackend: () => Promise<void>;
```

Comment:
loadUserFromBackend in the state just is duplicating information you already have in the props. I see no reason to include it.

Reply:
Refactored this in the latest commit.